### PR TITLE
Improve typing for logging

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #7037

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @carmocca @edward-io @ananthsub @borda

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/logger_connector/result.py [hunk 1]
```diff
@@ -225,20 +227,23 @@ def update(self, value: _IN_METRIC, batch_size: int) -> None:
...
-                self.cumulated_batch_size += batch_size
+                # `Metric.add_state` does not work well with mypy, mypy doesn't know this is a `Tensor`
+                # we could add an assertion, but this is a hot code path
+                self.cumulated_batch_size += batch_size  # type: ignore[operator]
             elif self.meta.is_max_reduction or self.meta.is_min_reduction:
                 self.value = self.meta.reduce_fx(self.value, value.mean())
             elif self.meta.is_sum_reduction:
                 self.value += value.mean()
+            else:
+                raise ValueError(f"Invalid `reduce_fx`: {self.meta.reduce_fx!r}")
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/connectors/logger_connector/result.py:246]: I think we already have a MisconfigException when we initialize meta.

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/logger_connector/result.py (hunk 1 above)
Comment (justusschock) [reply pytorch_lightning/trainer/connectors/logger_connector/result.py:246]: but for mypy there needs to be an `else` I think

**Review by justusschock — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -59,8 +59,7 @@ module = [
     "pytorch_lightning.loops.optimization.closure.py",
     "pytorch_lightning.loops.optimization.manual_loop.py",
     "pytorch_lightning.loops.evaluation_loop",
-    "pytorch_lightning.trainer.connectors.logger_connector.py",
-    "pytorch_lightning.trainer.connectors.logger_connector.fx_validator.py",
+    "pytorch_lightning.trainer.connectors.logger_connector.*",
```
Comment (awaelchli) [inline pyproject.toml:62]: I think the list changed on master and now you will have to remove the logger_connector from the new list instead of adding it

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**